src/lib/mina_base/transaction_logic.ml

```diff
@@ -1451,7 +1460,7 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
         ; use_full_commitment && not is_start
         ]
       |> Result.ok_if_true
-           ~error:Transaction_status.Failure.Update_not_permitted
+           ~error:Transaction_status.Failure.Parties_replay_check_failed
```

Comment:
This message for this failure seems mismatched with the check here.

Also, it feels like this check is wrong in general, because we want to be able to run snapps without worrying about the nonce or using the full commitment, and those snapps may use a non-empty predicate

Reply:
We won't be able to detect replays in such cases, right?
cc @A-Manning

Reply:
AFAIK that's correct @deepthiskumar